Eatery/Models/Eatery.swift

```diff
@@ -200,22 +200,20 @@ extension Eatery {
 
 extension Eatery {
 
-    var isFavorite: Bool {
-        get {
-            let ar = UserDefaults.standard.stringArray(forKey: "favorites") ?? []
-            return ar.contains { $0 == name }
-        }
-        set {
-            var ar = UserDefaults.standard.stringArray(forKey: "favorites") ?? []
-            if newValue {
-                ar.append(name)
-            } else {
-                ar.removeAll(where: { $0 == name })
-            }
-            UserDefaults.standard.set(ar, forKey: "favorites")
+    func isFavorite() -> Bool {
+        return UserDefaults.standard.stringArray(forKey: "favorites")?.contains(name) ?? false
+    }
 
-            NotificationCenter.default.post(name: .eateryIsFavoriteDidChange, object: self)
+    func setFavorite(_ newValue: Bool) {
+        var ar = UserDefaults.standard.stringArray(forKey: "favorites") ?? []
```

Comment:
can we call this `var favorites`

Reply:
Turns out we can't because if we have `var favorites` then we have to sacrifice immutability in some other classes that modify `isFavorite`, even though `isFavorite` is not a stored property

The options (that I could think of) were:
- Have it be `var favorites` and use `var eatery: Eatery` everywhere
- Have it be like it is above
- Have Eatery be a class